src/client/client.go

```diff
@@ -545,6 +546,30 @@ func NewEnterpriseClientOnUserMachine(prefix string, options ...Option) (*APICli
 	return newOnUserMachine(cfg, context, name, prefix, options...)
 }
 
+func CheckPortForwardLiveness(context *config.Context) bool {
+	// Check if old PortForwarder is still running. Note: 'pid' will not be
+	// set by pachctl <2.0.4, so it won't always exist.
+	if pid, ok := context.PortForwarders["pid"]; ok {
+		// If old port-forward pid no longer exists at all, it's dead.
+		if _, err := os.Stat(fmt.Sprintf("/proc/%d", pid)); errors.Is(err, os.ErrNotExist) {
+			return false
+		} else if err != nil {
+			log.Errorf("Can't stat /proc/%d to establish whether old port-forward process is alive: %v", pid, err)
+			return true // assume alive
+		}
+		// If old port-forward pid isn't "pachctl port-forward", it's dead.
+		if pfname, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/cmdline", pid)); err == nil {
+			if !bytes.HasPrefix(pfname, []byte{'p', 'a', 'c', 'h', 'c', 't', 'l'}) {
```

Comment:
```suggestion
			if !strings.HasPrefix(string(pfname), "pachctl") {
```
looks a little nicer to me, maybe there's some reason to avoid it

Reply:
No real reason, LGTM